src/pages/dar_application/DataAccessRequest_new.js

```diff
@@ -0,0 +1,330 @@
+import { useState, useEffect, useCallback} from 'react';
+import { a, div, h, h4, span, p } from 'react-hyperscript-helpers';
+import isNil from 'lodash/fp/isNil';
+import { DataSet, DAR } from '../../libs/ajax';
+import { FormField, FormFieldTitle, FormFieldTypes, FormValidators } from '../../components/forms/forms';
+
+const searchOntologies = (query, callback) => {
+  let options = [];
+  DAR.getAutoCompleteOT(query).then(
+    items => {
+      options = items.map(function(item) {
+        return item.label;
+      });
+      callback(options);
+    });
+};
+
+const searchDatasets = (query, callback) => {
+  DataSet.searchDatasets(query).then(items => {
+    let options = items.map(function (item) {
+      return formatSearchDataset(item);
+    });
+    callback(options);
+  });
+};
+
+const formatSearchDataset = (ds) => {
+
+  return {
+    key: ds.dataSetId,
+    value: ds.dataSetId,
+    dataset: ds,
+    displayText: ds.datasetIdentifier,
+    label: span({}, [
+      span({style: {fontWeight: 'bold'}}, [ds.datasetIdentifier]), ' | ', ds.name]),
+  };
+};
+
+const datasetsContainDataUseFlag = (datasets, flag) => {
+  return datasets.some((ds) => ds?.dataUse[flag] === true);
+
+};
+
+const fetchAllDatasets = async (dsIds) => {
+  let datasets;
+  if (!isNil(dsIds)) {
+    datasets = await Promise.all(dsIds.map((id) => DataSet.getDatasetByIdV2(id)));
```

Comment:
would it be worth changing the get all datasets endpoint to allow filtering to ids, so we don't have to do many calls? e.g., `/api/v2/dataset?id=1,2,3`

Reply:
I think that's a great idea for a new endpoint. We try not to modify in-use APIs when we can just create a new one and deprecate the old.

Reply:
okay ! do you think this best be `/api/v3/dataset` or something under `/api/v2/dataset/...`?

Reply:
Not sure about either one ... Justin just did something similar here: https://github.com/DataBiosphere/consent/pull/1564
Maybe `GET /api/datasets?ids=id1,id2,id3` fits the bill?

Reply:
Oh, I like his idea of `/api/dataset/batch?ids=1,2,3,4`. I will write up a ticket with that (subject to change) and implement it.

Reply:
https://github.com/DataBiosphere/consent/pull/1780 <-- once this is merged I can update this GET call.

Reply:
added into this PR